apps/frontend/src/app/5_pages/ProposalPage/hooks/useGetProposals.ts

```diff
@@ -0,0 +1,15 @@
+import { useMemo } from 'react';
+
+import { areAddressesEqual } from '../../../../utils/helpers';
+import { useGetProposals } from '../../BitocracyPage/hooks/useGetProposals';
+
+export const useGetProposalById = (id: string | undefined) => {
+  const { loading, data } = useGetProposals();
+
+  const proposal = useMemo(
+    () => data.find(item => areAddressesEqual(item.id, id || '')),
+    [data, id],
+  );
+
+  return { loading, proposal };
```

Comment:
Use a subgraph query which actually queries for a single proposal by it's id, list is unnecessary for this page. You can then request additional fields that are needed here (like emittedBy.majorityPercentageVotes and etc)